### PR TITLE
Fix broken config handling SLE 12 GA

### PR DESCRIPTION
**Pull request by imobach**
This patch should fix bsc#954494 adding some basic tests. See https://github.com/yast/yast-slp-server/pull/12

### CODE REVIEW

**Comment by mvidner** [comment]
OK, this is a test of the `Read` method.

I was actually thinking of whitebox-testing the parser regex, writing a file using all variations, like
```
#foo bar
#foo  bar
#foo=bar
#foo = bar
#foo =bar
#foo= bar

;foo bar
;foo  bar
;foo=bar
;foo = bar
;foo =bar
;foo= bar

foo bar
foo  bar
foo=bar
foo = bar
foo =bar
foo= bar
```
and making sure it parses as expected

File: src/scrconf/slp_conf.scr [hunk 1]
```diff
@@ -37,7 +37,7 @@
         // we need to exclude ; because of the second matching rule
         "params" : [
                 // Options with one value ('yes' / 'no')
-                $[ "match" : [ "^[#;][ \t]*([^ \t\=]+)[ \t\=]+[ ]*(.+)[ \t]*$", ";%s = %s" ]],
+                $[ "match" : [ "^[#;][ \t]*([^ \t\=]+)[ \t\=]*[ ]*(.+)[ \t]*$", ";%s = %s" ]],
                 // Options with more possible values
```
Comment (mvidner) [inline src/scrconf/slp_conf.scr]: Now, before this fix, this 1st element of "params" differed from the 2nd only by the comment character: `^[#;]...` vs `^...`. Shouldn't the 2nd one be changed too?

**Comment by imobach** [comment]
Ok, I like the idea and it shouldn't be that hard.

**Comment by mvidner** [comment]
I may be overcomplicating this, but only now I noticed that there is another file with a different but similar parser definition: [slp_reg.scr](https://github.com/imobach/yast-slp-server/blob/fix-broken-config-handling-sle12-ga/src/scrconf/slp_reg.scr). Does it perhaps have a similar bug?

**Comment by imobach** [comment]
I've added some tests to check that both expressions are working. Regarding the second one, it does not seem to be affected by the bug (but I could make the same change as in the first one just consistency).

Regarding `slp_reg.scr` I need to check first whether it is affected by the original bug.

**Comment by imobach** [comment]
@mvidner About `slp_reg.scr` they could be affected by the same bug, but there are some differences. First of all, files under `/etc/slp_reg.d` won't contain lines commented with `;`. And the only way to have such lines is modifying the file by hand.

But the weird thing is that, if you comment one of those lines by hand, the `slp-server` module will show them in the UI as if they were "uncommented". And from my POV that's another bug.

File: test/scr_parse_test.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+RSpec.describe "slp.conf parsing" do
+  COMMENTED_LINES = File.join(FIXTURES_PATH, "commented-slp.conf")
+  UNCOMMENTED_LINES = File.join(FIXTURES_PATH, "uncommented-slp.conf")
+
+  COMMENTED_REGEXP = Regexp.new '^[#;][ \t]*([^ \t\=]+)[ \t\=]*[ ]*(.+)[ \t]*$'
+  UNCOMMENTED_REGEXP = Regexp.new '[ \t]*([^ \t\=]+)[ \t\=]+[ ]*(.+)[ \t]*$'
+
+  File.readlines(COMMENTED_LINES).each do |line|
+    it "extracts param name and value from '#{line.strip}'" do
```
Comment (mvidner) [inline test/scr_parse_test.rb]: Better `line.chomp` to avoid stripping spaces, just to future-proof.

File: test/scr_parse_test.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+RSpec.describe "slp.conf parsing" do
+  COMMENTED_LINES = File.join(FIXTURES_PATH, "commented-slp.conf")
+  UNCOMMENTED_LINES = File.join(FIXTURES_PATH, "uncommented-slp.conf")
+
+  COMMENTED_REGEXP = Regexp.new '^[#;][ \t]*([^ \t\=]+)[ \t\=]*[ ]*(.+)[ \t]*$'
+  UNCOMMENTED_REGEXP = Regexp.new '[ \t]*([^ \t\=]+)[ \t\=]+[ ]*(.+)[ \t]*$'
```
Comment (mvidner) [inline test/scr_parse_test.rb]: Aw, this duplication of the regexp is quite error prone I fear. Not to mention subtle differences between C and Ruby regexes. I think I can make a test that uses the production parser.

Comment (imobach) [inline test/scr_parse_test.rb]: Yes, I agree: I didn't know how to use the original parser. Any idea?

File: test/scr_parse_test.rb (hunk 2 above)
Comment (imobach) [inline test/scr_parse_test.rb]: You're right. Done.

File: test/scr_parse_test.rb (hunk 3 above)
Comment (mvidner) [inline test/scr_parse_test.rb]: I think this should work: move `test/fixtures/{un,}commented-slp.conf` to `test/fixtures/{un,}commented}/etc/slp.conf`, that is, treat them like the `valid` and `broken` cases, then `change_scr_root(fixtures); SCR.Read(path(".etc.slp.all"))` and check that.

Comment (imobach) [inline test/scr_parse_test.rb]: Thanks for the hint. I'll try it.

**Comment by mvidner** [comment]
LGTM.
Thanks for putting the crazy parser under control with these tests!

**Comment by imobach** [comment]
Thanks for your guidance :)

**Pull request merged**